core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -194,35 +212,37 @@ class TransactionCoordinator(brokerId: Int,
                                        pid: Long,
                                        epoch: Short,
                                        partitions: collection.Set[TopicPartition],
-                                       responseCallback: TxnMetadataUpdateCallback): Unit = {
+                                       responseCallback: AddPartitionsCallback): Unit = {
     val errors = validateTransactionalId(transactionalId)
     if (errors != Errors.NONE)
       responseCallback(errors)
     else {
       // try to update the transaction metadata and append the updated metadata to txn log;
       // if there is no such metadata treat it as invalid pid mapping error.
-      val (error, newMetadata) = txnManager.getTransactionState(transactionalId) match {
+      val (error: Errors, coordinatorEpoch: Int, txnMetadata: TransactionMetadata) = txnManager.getTransactionState(transactionalId) match {
```

Comment:
Btw, a better way to represent this would be using `Either[Errors, (Int, TransactionMetadata)]`. Even better would be to have a type for the tuple on the right, but even without it, it would make the flow way clearer IMO.